### PR TITLE
[11.0][ADD] web_view_calendar_list

### PR DESCRIPTION
**Pull request by etobella**
This module adds a new view type that can be used to show calendars as lists.

Usage
=====

Create a new view using the calendar_list tag.
It has the same options than calendar::

    <calendar_list date_start="schedule_date" date_delay="duration" color="partner_id">
        <field name="display_name"/>
        <field name="partner_id"/>
    </calendar_list>
image

@jarroyomorales @JordiBForgeFlow

### CODE REVIEW

File: web_view_calendar_list/static/src/js/calendar_list_controller.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+odoo.define('web_view_calendar_list.CalendarListController', function (require) {
```
Comment (jarroyomorales) [inline web_view_calendar_list/static/src/js/calendar_list_controller.js:1]: Is this file necessary?

File: web_view_calendar_list/views/webclient_templates.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+    <template id="assets_backend"
+              name="Backend Assets (used in backend interface)"
+              inherit_id="web.assets_backend">
+        <xpath expr="." position="inside">
+            <script type="text/javascript" src="/web_view_calendar_list/static/src/js/calendar_list_renderer.js"/>
+            <script type="text/javascript" src="/web_view_calendar_list/static/src/js/calendar_list_model.js"/>
+            <script type="text/javascript" src="/web_view_calendar_list/static/src/js/calendar_list_controller.js"/>
+            <script type="text/javascript" src="/web_view_calendar_list/static/src/js/calendar_list_view.js"/>
+            <!--<link rel="stylesheet" type="text/less" href="/medical_surgical_appointment/static/src/less/appointment.less"/>-->
```
Comment (jarroyomorales) [inline web_view_calendar_list/views/webclient_templates.xml]: Remove line

File: web_view_calendar_list/static/src/js/calendar_list_controller.js (hunk 1 above)
Comment (etobella) [reply web_view_calendar_list/static/src/js/calendar_list_controller.js:1]: I made it just in case

File: web_view_calendar_list/static/src/js/calendar_list_renderer.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                },
+                height: 'parent',
+                unselectAuto: false,
+                locale: locale, // reset locale when fullcalendar has already been instanciated before now
+            });
+
+            this.$calendar.fullCalendar(fc_options);
+        },
+
+        _render: function () {
```
Comment (AdriaGForgeFlow) [inline web_view_calendar_list/static/src/js/calendar_list_renderer.js:91]: It's a pity that you need to override the whole function just to change the last line to prevent it from calling _super but there is no other option right?

**Review by AdriaGForgeFlow — APPROVED** [review]
Works like a charm :+1:

File: web_view_calendar_list/static/src/js/calendar_list_renderer.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
+odoo.define('web_view_calendar_list.CalendarListRenderer', function (require) {
+    "use strict";
+
+    var CalendarRenderer = require('web.CalendarRenderer');
+    var session = require('web.session');
+    var core = require('web.core');
+    var qweb = core.qweb;
+    var _t = core._t;
+
+    var scales = {
```
Comment (etobella) [inline web_view_calendar_list/static/src/js/calendar_list_renderer.js:10]: @AdriaGForgeFlow The main reason for the function is this. We need to rewrite a constant item :cry:

File: web_view_calendar_list/static/src/js/calendar_list_controller.js (hunk 1 above)
Comment (etobella) [reply web_view_calendar_list/static/src/js/calendar_list_controller.js:1]: Also, I think it is a good approach to define an specific class, but it is not used right now.

**Review by jarroyomorales — APPROVED** [review]
LGTM! :+1:

**Comment by etobella** [comment]
@OCA/web-maintainers

File: web_view_calendar_list/static/src/js/calendar_list_renderer.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+                eventRender: function (event, element) {
+                    var $render = $(self._eventRender(event));
+                    event.title = $render.find('.o_field_type_char:first').text();
+                    element.find('.fc-list-item-title').html($render.html());
+                    element.addClass($render.attr('class'));
+                    var display_hour = '';
+                    if (!event.allDay) {
+                        var start = event.r_start || event.start;
+                        var end = event.r_end || event.end;
+                        var timeFormat = _t.database.parameters.time_format.search("%H") != -1 ? 'HH:mm': 'h:mma';
```
Comment (simahawk) [inline web_view_calendar_list/static/src/js/calendar_list_renderer.js:56]: can't you use format utils?

File: web_view_calendar_list/static/src/js/calendar_list_renderer.js (hunk 3 above)
Comment (simahawk) [reply web_view_calendar_list/static/src/js/calendar_list_renderer.js:91]: @etobella can you explain w/ a detailed comment?

File: web_view_calendar_list/views/webclient_templates.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (simahawk) [inline web_view_calendar_list/views/webclient_templates.xml]: could you move this to `templates/assets.xml`?

**Review by simahawk — CHANGES_REQUESTED** [review]
@etobella nice addition, thanks!
A bunch of comments on my side :stuck_out_tongue_winking_eye:

File: web_view_calendar_list/static/src/js/calendar_list_renderer.js (hunk 3 above)
Comment (etobella) [reply web_view_calendar_list/static/src/js/calendar_list_renderer.js:91]: The main reason why we need to overwrite all is the variables `scales`. It is defined as an external variable undependant on the class, so, in order to modify it, we need to overwrite all the function :sob:

Comment (etobella) [reply web_view_calendar_list/static/src/js/calendar_list_renderer.js:91]: I will add the comment on the JS

File: web_view_calendar_list/static/src/js/calendar_list_renderer.js (hunk 5 above)
Comment (etobella) [reply web_view_calendar_list/static/src/js/calendar_list_renderer.js:56]: The field is a datetime, and no parser exists on field_utils in order to parse datetime to time. I am following the same behavior found on odoo. It might not be the best, but I tried to follow the same logic in order to not make much changes.

File: web_view_calendar_list/static/src/js/calendar_list_renderer.js (hunk 3 above)
Comment (Tardo) [reply web_view_calendar_list/static/src/js/calendar_list_renderer.js:91]: Maybe you can modify methods where '_render' is called and pass as argument the new "scales". "Hoisting" do the rest of the magic... :/

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
@etobella can you check the builds?

**Comment by etobella** [comment]
Sorry, a misspell now it should be :green_apple:
